sympy/ntheory/digits.py

```diff
@@ -17,12 +17,37 @@ def digits(n, b=10):
     >>> from sympy.ntheory.digits import digits
     >>> digits(35)
     [10, 3, 5]
-    >>> digits(27, 2)
+    >>> digits(27, 2, None)
```

Comment:
no need to change existing tests, just add a single easily interpreted test in base 10 that uses bits greater than needed like `digits(35, 10, 4) -> [10, 0, 0, 3, 5]`. And rather than allow more digits than requested to be returned silently, I think we should use an OverflowError to indicate that the digits exceeded the space allowed, e.g. `digits(35, 10, 1) -> OverflowError`. `ibin` could include the same.

Reply:
I have already pushed the tests. Never mind.